src/reader.rs

```diff
@@ -95,28 +114,89 @@ impl LogReceiver {
 
     fn new_sender(&self) -> LogSender {
         LogSender {
-            _send_counter: self.send_counter.clone(),
+            send_guard: self.send_guard.clone(),
             send_handle: self.send_handle.clone(),
         }
     }
+
+    fn kill_senders(&self) {
+        self.send_guard.is_receiver_closed.store(true, Ordering::Release);
+        while self.send_guard.active_sender_count.load(Ordering::Acquire) != 0 { }
```

Comment:
Why is this loop necessary?

Reply:
The idea is to prevent the following (very unlikely) chain of events:
* In thread A: `is_receiver_closed` check passes in `synchronize_send`.
* In thread B: `stop_log_senders` is called, then all messages currently in the `Receiver` are read and logged.
* In thread A: A message is sent to the `Receiver`.
* In thread B: The `Receiver` is dropped. (too late)

I could also do this with a `RwLock<()>`, I suppose, which may be a cleaner solution. Is adding `parking_lot` as a dependency a possibility?